app/models/workflow/instance/step.rb

```diff
@@ -40,11 +40,31 @@ def is_manual?
 
     private
 
-    def update_completed_counter_cache
+    def update_process
+      update_process_completed_counter_cache
+      update_process_completion_status
+    end
+
+    def update_process_completed_counter_cache
       self.process.completed_steps_count = Workflow::Instance::Step.where(completed: true, process_id: process.id).count
       self.process.save
     end
 
+    def update_process_completion_status
```

Comment:
caching the completion status of a process, otherwise this new api endpoint takes too long to run.